### PR TITLE
flamenco, fuzz: (syscall fuzzing) load input data regions correctly

### PR DESCRIPTION
**Pull request by ravyu-jump**
Loads input regions correctly, reflecting write permissions

### CODE REVIEW

File: src/flamenco/runtime/tests/fd_exec_instr_test.c [hunk 1]
```diff
@@ -1665,41 +1665,28 @@ fd_exec_vm_syscall_test_run( fd_exec_instr_test_runner_t * runner,
...
-    if( !input->vm_ctx.input_data_regions[i].content ) {
+  /* Load input data regions */
+  fd_vm_input_region_t * input_regions = fd_valloc_malloc( valloc, alignof(fd_vm_input_region_t), sizeof(fd_vm_input_region_t) * input->vm_ctx.input_data_regions_count );
+  ulong input_data_offset = 0UL;
+  for( ulong i=0; i<input->vm_ctx.input_data_regions_count; i++ ) {
+    fd_exec_test_input_data_region_t const * region = &input->vm_ctx.input_data_regions[i];
+    pb_bytes_array_t * array = region->content;
+    if( !array ) {
       continue;
     }
```
Comment (ravyu-jump) [inline src/flamenco/runtime/tests/fd_exec_instr_test.c:1676]: @ibhatt-jumptrading are empty regions (ie., region_sz 0) possible? Should I not `continue`?

Comment (topointon-jump) [reply src/flamenco/runtime/tests/fd_exec_instr_test.c:1676]: It looks like they are possible (https://github.com/anza-xyz/agave/blob/3072c1a72b2edbfa470ca869f1ea891dfb6517f2/programs/bpf_loader/src/serialization.rs#L105)

Comment (topointon-jump) [reply src/flamenco/runtime/tests/fd_exec_instr_test.c:1676]: https://github.com/anza-xyz/agave/blob/3072c1a72b2edbfa470ca869f1ea891dfb6517f2/programs/bpf_loader/src/serialization.rs#L136 they are not possible

File: src/flamenco/runtime/tests/fd_exec_instr_test.c [hunk 2]
```diff
@@ -1665,41 +1665,28 @@ fd_exec_vm_syscall_test_run( fd_exec_instr_test_runner_t * runner,
...
-    if( !input->vm_ctx.input_data_regions[i].content ) {
-      continue;
-    }
-    fd_memcpy( input_data_ptr, array->bytes, array->size );
-    input_data_ptr += array->size;
-  }
-  if( input_data_ptr != (input_data + input_data_sz) ) {
-    goto error;
+    input_regions[i].vaddr_offset = input_data_offset; // Follow solfuzz-agave convention instead of using region->offset
+    input_regions[i].haddr = (ulong)array->bytes; // TODO: Check if lifetime is an issue
```
Comment (topointon-jump) [inline src/flamenco/runtime/tests/fd_exec_instr_test.c:1678]: remove comment

**Review by topointon-jump — APPROVED** [review]

**Review by topointon-jump — APPROVED** [review]

**Review by topointon-jump — APPROVED** [review]

**Pull request merged**